core/src/main/java/org/elasticsearch/transport/netty/NettyTransportChannel.java

```diff
@@ -134,6 +141,14 @@ public void sendResponse(Throwable error) throws IOException {
         transportServiceAdapter.onResponseSent(requestId, action, error);
     }
 
+    private void close() {
```

Comment:
In my original implementation it was called indeed in the `finally` block, but Boaz and I had a lengthy discussion about this one. It makes sense to free the reserved bytes at the beginning of the method as we're done with request processing at this point in time. Freeing at the beginning guards against any problems that might occur further down the line (admittedly this could also be solved in a finally block).

The name `close()` just implies that you cannot reuse the channel after this method has been called and the precondition check guards against this.